### PR TITLE
Change message image when speaking with the bot or the agent

### PR DESCRIPTION
**Pull request by Iru89**
## Description
To be able to define a different image for the messages sent by the agent

## Context
Until now we could not define a different image for the messages sent by the agent, what we did was to change all the images of the chat while the user was talking with an agent, but when the handoff was finished all the messages had the image of the bot again.

## Approach taken / Explain the design
- an image can be defined in the custom property `agentMessageImage`
- this is used when a message arrives with the sentBy = agent
- the css class of the timestamp will be `botanic-timestamp-agent` to be able to define another css or another image in that div (for now this is done with css, in the future it would be interesting to be a custom component)

### CODE REVIEW

File: packages/botonic-react/src/components/message/timestamps.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    getFormattedTimestamp,
+    timestampStyle,
+  }
+}
+
+export const MessageTimestamp = ({
+  timestamp,
+  style,
+  isSentByUser,
+  sentBy,
```
Comment (AinaVendrell) [inline packages/botonic-react/src/components/message/timestamps.jsx:48]: do we need this to prop? can we not get `isSentByUser` ny using the `sentBy` prop

File: packages/botonic-react/src/components/message/message-image.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React, { useContext } from 'react'
+
+import { WEBCHAT } from '../../constants'
+import { WebchatContext } from '../../contexts'
+import { resolveImage } from '../../util/environment'
+import { BotMessageImageContainer } from './styles'
+
+interface MessageImageProps {
+  imagestyle: Record<string, unknown>
```
Comment (AlbertGom) [inline packages/botonic-react/src/components/message/message-image.tsx:9]: camelCase: imageStyle?

File: packages/botonic-react/src/components/message/message-image.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  isSentByUser: boolean
+  isSentByAgent: boolean
```
Comment (AlbertGom) [inline packages/botonic-react/src/components/message/message-image.tsx:11]: I think it's better to use a variable called `sentBy` that is an enum that can have values 'agent', 'bot' and 'user'?

File: packages/botonic-react/src/components/message/timestamps.jsx (hunk 1 above)
Comment (AlbertGom) [reply packages/botonic-react/src/components/message/timestamps.jsx:48]: Agree with Aina

**Review by AlbertGom — APPROVED** [review]

**Comment by AlbertGom** [comment]
@Iru89 Try to add tests for this new behaviour!

**Comment by AlbertGom** [comment]
Looks good :clap:

File: packages/botonic-react/src/components/message/timestamps.jsx (hunk 1 above)
Comment (Iru89) [reply packages/botonic-react/src/components/message/timestamps.jsx:48]: ok I will remove the isSentByUser

File: packages/botonic-react/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@botonic/react",
-  "version": "0.24.0-alpha.2",
+  "version": "0.24.0-alpha.4",
```
[the remark this answers is not in the document]
Comment (Iru89) [reply packages/botonic-react/package.json:3]: The alpha.3 already existed

File: packages/botonic-react/src/components/message/index.jsx [hunk 5]
```diff
@@ -241,20 +195,15 @@ export const Message = props => {
...
-    )
     const animationsEnabled = getThemeProperty(
       WEBCHAT.CUSTOM_PROPERTIES.enableAnimations,
       true
     )
 
     const resolveCustomTypeName = () =>
-      isSentByBot && type === INPUT.CUSTOM ? ` ${m.customTypeName}` : ''
+      isSentByBot && type === INPUT.CUSTOM
+        ? ` ${messageJSON.customTypeName}`
```
[the remark this answers is not in the document]
Comment (Iru89) [reply packages/botonic-react/src/components/message/index.jsx:205]: This space was already there before and I think it is necessary to show two classes of css, in the 208 line

File: packages/botonic-react/src/components/message/message-image.tsx (hunk 2 above)
Comment (Iru89) [reply packages/botonic-react/src/components/message/message-image.tsx:9]: No the correct prop is imagestyle. There is an old [PR#1063](https://github.com/hubtype/botonic/pull/1063) in the description of this PR that says it is better to use imagestyle. 
There is also a code that makes a log saying that legacy is imageStyle and the new imagestyle (src/util/logs.js).

File: packages/botonic-react/src/components/message/message-image.tsx (hunk 3 above)
Comment (Iru89) [reply packages/botonic-react/src/components/message/message-image.tsx:11]: Yes, I will refactor it to pass only sentBy and do the check inside the component.

**Pull request merged**